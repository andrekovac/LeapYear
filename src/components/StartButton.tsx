import React from "react";
import styled from "styled-components/native";

import Text from "./Text";

import { paddingNormal } from '../constants/style';

type StartButtonProps = {
  text: string;
  onPress: () => void;
};

const StartButton = ({ text, onPress }: StartButtonProps) => (
  <Button onPress={onPress}>
    <TextWrapper>
      <Text>{text}</Text>
    </TextWrapper>
  </Button>
);

const Button = styled.TouchableOpacity`
  padding: 0 ${paddingNormal};
  width: 100%;
`;

const TextWrapper = styled.View`
  padding: ${paddingNormal} 0;
  background-color: white;
  border-radius: 35px;

  align-items: center;
`;

export default StartButton;
