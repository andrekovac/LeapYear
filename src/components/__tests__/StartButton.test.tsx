import React from "react";
import renderer from "react-test-renderer";

import StartButton from "../StartButton";

it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <StartButton text={"StartButton Snapshot Test"} onPress={() => {}} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
