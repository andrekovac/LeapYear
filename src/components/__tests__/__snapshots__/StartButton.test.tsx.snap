// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 35,
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "100",
          "textAlign": "center",
        }
      }
    >
      StartButton Snapshot Test
    </Text>
  </View>
</View>
`;
